pkg/unidling/controller/unidling_controller.go

```diff
@@ -99,10 +100,18 @@ func NewUnidlingController(scaleNS scale.ScalesGetter, mapper meta.RESTMapper, e
 
 	_, controller := cache.NewInformer(
 		&cache.ListWatch{
-			// No need to list -- we only care about new events
+			// https://bugzilla.redhat.com/show_bug.cgi?id=1887745 -- adding a limit on the amount of
+			// events listed from the api-server, to avoid crash by loading too may events at once.
 			ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+				fn := func(ctx context.Context, opts metav1.ListOptions) (runtime.Object, error) {
+					return evtNS.Events(metav1.NamespaceAll).List(ctx, opts)
+				}
+				p := pager.New(pager.ListPageFunc(fn))
+
 				options.FieldSelector = fieldSelector.String()
-				return evtNS.Events(metav1.NamespaceAll).List(context.TODO(), options)
+				options.Limit = 500
+				list, _, err := p.List(context.TODO(), options)
```

Comment:
Perhaps mark a TODO comment - we should wire a global context through to the controller.

Reply:
Sure. I'm adding a comment to highlight.